### DISCUSSION TITLE
Open WebUI and Context Window/LLM Memory

### DISCUSSION THREAD

**Discussion by rbrinson** [open-webui/open-webui#3576] [Q&A]
I've got a few questions concerning context window/LLM memory. I have this idea for a long running chat with a custom model in Open WebUI. It could last for weeks, perhaps even months. I would create a modelfile with a custom system prompt such that the model would take on a persona helpful to the type of work. The work would involve interactive creative writing but also referencing markdown documents for specific facts to help guide the work.

One of the features in the "Create a Model" form in Open WebUI is to Select Documents that have been previously uploaded. I have a bundle of around 1,000 .md documents, around 78 MB on disk, that I planned to upload and add to the "Knowledge" of this custom model. If I'm using dolphin-llama3 as the base for this custom model, I know that it has a context length of about 8k. I also know that the model system prompt will consume some of that context length. One of the questions that I had was does Open WebUI add the documents as part of the context window, or does the storage for those documents become part of the default search for results without affecting the context length being used for the chat?

The second question has to do with the Personalization | Memory within the User Settings. In order to facilitate a long running chat with a LLM, it will be necessary to help the LLM maintain memory of critical pieces of data. Otherwise, once the chat begins to overflow the context window, the LLM will begin to forget why we were even having this conversation. Does Open WebUI manage adding important memories to this feature, like [MemGPT](https://memgpt.ai/), or is that something that I maintain as needed? Also, do these stored Memories take up tokens in the context window, or are they part of a RAG database that can be searched without impacting context length?

Finally, I've also noticed in the General section of Settings, there is an Advanced Parameters section that includes variables that we can adjust, including Context Length. Given that the models have their own values for some of these parameters, what do those values affect? Can any of those parameters be manipulated to help facilitate longer running chats?

Thank you for any insight!

**Comment by JoshJarabek7** [comment]
Did you ever figure this out? I have the same issue/question.

**Comment by estcap2** [comment]
I have the same questions
